config/config.go

```diff
@@ -162,6 +163,23 @@ func setDefaultValues(conf *config) {
 		slog.Warn("Mongo DBName missing, using default", "value", defaultConfig.Mongo.DBName)
 	}
 
+	if(conf.MeteoProviders[0].Name==""){
+		conf.MeteoProviders[0].Name=defaultConfig.MeteoProviders[0].Name
+	}
+	if(conf.MeteoProviders[0].APIPath==""){
+		if(conf.MeteoProviders[0].Name=="open-meteo"){
+			conf.MeteoProviders[0].APIPath=defaultConfig.MeteoProviders[0].APIPath
+		}else { 
+			validateCriticalFields(conf);
```

Comment:
I personally don't thing you need to call validateCriticalFields again and again, as ur calling it anyways at line 127, and idk if we need a separate function for validating critical fields, @raamsri thoughts? But other than that, LGTM.

Reply:
we need a seperate function for validating critical feilds that would improve the code readability

My thought process for calling the function again was 

I am calling the function because I know something is wrong when the name is not open meteo and the api path is missing, I want to validate immediately coz a critical one is missing and I want to resolve asap.

Reply:
I see. I also noticed another thing while rereading ur code, ur only checking defaultConfig.MeteoProviders[0]. Shouldn't you loop there as well?

Reply:
As of now we have only one provider so did the 0 part.

When we get to multiple providers we do that ?

Reply:
> As of now we have only one provider so did the 0 part.
> 
> When we get to multiple providers we do that ?

Isn't it better to finish this off now itself? Why pile up the work..